core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/TransportSnapshotsStatusAction.java

```diff
@@ -196,12 +197,13 @@ private SnapshotsStatusResponse buildResponse(SnapshotsStatusRequest request, Li
         // Now add snapshots on disk that are not currently running
         if (Strings.hasText(request.repository())) {
             if (request.snapshots() != null && request.snapshots().length > 0) {
-                for (String snapshotName : request.snapshots()) {
-                    SnapshotId snapshotId = new SnapshotId(request.repository(), snapshotName);
-                    if (currentSnapshotIds.contains(snapshotId)) {
+                for (String name : request.snapshots()) {
+                    SnapshotName snapshotName = new SnapshotName(request.repository(), name);
+                    if (currentSnapshotNames.contains(snapshotName)) {
                         // This is a snapshot the is currently running - skipping
                         continue;
                     }
+                    final SnapshotId snapshotId = snapshotsService.resolveSnapshotId(snapshotName);
```

Comment:
The original implementation also ends up calling `repository#readSnapshot` which goes to disk to read snapshot info, so presumably with this operation executing in the generic thread pool, it won't cause any more blocking than the current implementation?